### PR TITLE
Make legacy template noop updates not change cluster state

### LINKED ISSUE #57662: Streamline no-op template updates
Today every put-template request (both composable and legacy) results in a cluster state update, even if the cluster state already contains exactly the template that we're trying to put. There's no need to update the cluster state if the template is already there, so it would be preferable to skip any such requests without needing to publish a new cluster state version.

### PR DESCRIPTION
**Pull request by dakrone**
When a legacy template does not change, there is no need to create and publish a new cluster state,
rather the update should be a noop.

Relates to #57851
Resolves #57662

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java [hunk 1]
```diff
@@ -727,7 +749,6 @@ public void testFindV2InvalidGlobalTemplate() {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/pull/57393")
```
Comment (andreidan) [inline server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java:730]: This seems to be related to composable templates, the mentioned PR was abandoned though. Is this really fixed? Or was this muted by accident initially?

**Review by andreidan — APPROVED** [review]
Thanks for fixing this Lee. 

LGTM

File: server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java (hunk 1 above)
Comment (dakrone) [reply server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java:730]: It was fixed in the past with Julie's mappings merging, the test was just never unmuted (it's fixed now though)

**Review by DaveCTurner — APPROVED** [review]
Thanks @dakrone, LGTM2

**Pull request merged**